File: Kotlin/src/K08.kt
import java.util.*
import kotlin.random.Random

class K08 : IExecutable() {
    override fun question_01() {
        val numbers = Vector<Int>()
        val reader = Scanner(System.`in`)
        try {
            for (i in 1..3) {
                print("${i}つ目の値を入力してください。＞")
                numbers.add(reader.nextInt())
            }
            println("どちらを調べますか？")
            print("（０：最大値　１：最小値）＞")
            val choice = reader.nextInt()
            when (choice) {
                0 -> println("${numbers.size}の中で最大値は${GetMaxValue(numbers[0], numbers[1], numbers[2])}です。")
                1 -> println("${numbers.size}の中で最小値は${GetMinValue(numbers[0], numbers[1], numbers[2])}です。")
            }
        } catch (ex: Exception) {
            System.err.println(ex.message)
        }
    }

    override fun question_02() {
        try {
            println("冒険が今始まる！")
            var playerHp = 200 + Random.nextInt(100)
            var continueGame = 1
            val reader = Scanner(System.`in`)
            while (continueGame != 0) {
                if (playerHp == 0)
                    return
                println("あなたのHP：$playerHp")
                print("奥に進みますか？（１：奥に進む　０．帰る）＞")
                continueGame = reader.nextInt()
                if (continueGame == 1) {
                    val golemLv = Random.nextInt(1, 4)
                    playerHp = AttackGolem(golemLv, playerHp)
                }
            }
            println("リレ〇ト！")
        } catch (ex: java.lang.Exception) {
            System.err.println(ex.message)
        }
    }

    override fun question_03() {
        return
    }

    override fun question_04() {
        return
    }
}

File: CSharp/k08.cs
using System;
using System.Collections.Generic;

namespace CSharp
{
    class K08 : IExecutable
    {
        public override void Question01()
        {
            List<int> numbers = new List<int>();


            try
            {
                for (int i = 0; i < 3; i++)
                {
                    Console.Write(string.Format("{0}つ目の値を入力してください。＞", i + 1));
                    numbers.Add(int.Parse(Console.ReadLine()));
                }

                Console.WriteLine("どちらを調べますか？");
                Console.Write("（０：最大値　１：最小値）＞");
                int choice = int.Parse(Console.ReadLine());

                switch (choice)
                {
                    case 0:
                        Console.WriteLine(
                            string.Format("{0}の中で最大値は{1}です。",
                            numbers.Count,
                            Utilities.GetMaxValue(numbers[0], numbers[1], numbers[2])));
                        break;
                    case 1:
                        Console.WriteLine(
                            string.Format("{0}の中で最小値は{1}です。",
                            numbers.Count,
                            Utilities.GetMinValue(numbers[0], numbers[1], numbers[2])));
                        break;
                    default:
                        break;
                }
            }
            catch (Exception ex)
            {
                Console.Error.WriteLine(ex.Message);
            }
        }

        public override void Question02()
        {
            try
            {
                Console.WriteLine("冒険が今始まる！");
                Random random = new Random();
                int playerHp = 200 + random.Next(0, 101);
                bool continueGame = true;
                do
                {
                    if (playerHp == 0)
                        return;

                    Console.WriteLine("あなたのHP：" + playerHp);
                    Console.Write("奥に進みますか？（１：奥に進む　０．帰る）＞");
                    continueGame = Convert.ToBoolean(int.Parse(Console.ReadLine()));
                    if (continueGame)
                    {
                        int golemLv = random.Next(0, 5);
                        Utilities.AttackGolem(golemLv, ref playerHp);
                    }

                } while (continueGame);
                Console.WriteLine("リレ〇ト！");
            }
            catch (Exception ex)
            {

                Console.Error.WriteLine(ex.Message);
            }
        }

        public override void Question03()
        {
            return;
        }

        public override void Question04()
        {
            return;
        }
    }
}
